pkg/volume/volume.go

```diff
@@ -43,7 +45,21 @@ func MountVolume(devicePath, mountPath, fsType string, mountOptions []string) er
 		return nil
 	}
 
-	mounter := &mount.SafeFormatAndMount{Interface: mount.New(""), Exec: mount.NewOSExec()}
+	// https://github.com/longhorn/longhorn/issues/2991
+	// pre v1.2 we ignored the fsType and always formatted as ext4
+	// after v1.2 we include the user specified fsType to be able to
+	// mount priorly created volumes we need to switch to the existing fsType
+	diskFormat, err := GetDiskFormat(devicePath)
+	if err != nil {
+		return err
+	}
+
+	// `unknown data, probably partitions` is used when the disk contains a partition table
+	if diskFormat != "" && !strings.Contains(diskFormat, "unknown data") && fsType != diskFormat {
```

Comment:
Should we error out if there are partitions detected in the Longhorn volume?

Reply:
The mount will either succeed or fail as part of the kubernetes mount call, no need to error here. Since this is just to determine if we need the backwards compatibility.